### PR TITLE
Adds topic selections for subscribers

### LINKED ISSUE #93: Cataloger ERROR: value too long for type character varying(255)
I think this is happening writing very long Institution Names.

Sample SCOAP record that demonstrates this problem:
http://repo.scoap3.org/record/9107/files/main.xml

```
in process about to evaluate: //datafield[@tag='700']/subfield[(@code='v')]
keyParts0: //datafield[@tag='700']/subfield[(@code='v')]
Post eval num hits: 2899

[[SNIP a few thousand lines of dumped Institution Names]]]

doc2p: fname: export/xm?ln=en
IDHits size: 2899
[ERROR] [03/11/2015 13:39:12.418] [application-akka.actor.default-dispatcher-6] [akka://application/user/cataloger] ERROR: value too long for type character varying(255)
org.postgresql.util.PSQLException: ERROR: value too long for type character varying(255)
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2270)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1998)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:570)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:420)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeUpdate(AbstractJdbc2Statement.java:366)
	at com.jolbox.bonecp.PreparedStatementHandle.executeUpdate(PreparedStatementHandle.java:205)
	at anorm.Sql$class.execute1(Anorm.scala:321)
	at anorm.SimpleSql.execute1(Anorm.scala:190)
	at anorm.Sql$class.executeInsert(Anorm.scala:351)
	at anorm.SimpleSql.executeInsert(Anorm.scala:190)
	at models.Topic$$anonfun$create$1.apply(Topic.scala:124)
	at models.Topic$$anonfun$create$1.apply(Topic.scala:118)
	at play.api.db.DBApi$class.withConnection(DB.scala:81)
	at play.api.db.BoneCPApi.withConnection(DB.scala:267)
	at play.api.db.DB$$anonfun$withConnection$3.apply(DB.scala:162)
	at play.api.db.DB$$anonfun$withConnection$3.apply(DB.scala:162)
	at scala.Option.map(Option.scala:145)
	at play.api.db.DB$.withConnection(DB.scala:162)
	at models.Topic$.create(Topic.scala:118)
	at workers.Cataloger.createTopic(Cataloger.scala:177)
	at workers.Cataloger$$anonfun$topics$1$$anonfun$apply$1.apply(Cataloger.scala:82)
	at workers.Cataloger$$anonfun$topics$1$$anonfun$apply$1.apply(Cataloger.scala:78)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at workers.Cataloger$$anonfun$topics$1.apply(Cataloger.scala:78)
	at workers.Cataloger$$anonfun$topics$1.apply(Cataloger.scala:70)
	at scala.collection.immutable.List.map(List.scala:272)
	at workers.Cataloger.topics(Cataloger.scala:70)
	at workers.Cataloger$$anonfun$catalog$2.apply(Cataloger.scala:306)
	at workers.Cataloger$$anonfun$catalog$2.apply(Cataloger.scala:306)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at workers.Cataloger$.catalog(Cataloger.scala:306)
	at workers.CatalogWorker$$anonfun$receive$1.applyOrElse(Cataloger.scala:33)
	at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
	at workers.CatalogWorker.aroundReceive(Cataloger.scala:31)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
	at akka.actor.ActorCell.invoke(ActorCell.scala:487)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
	at akka.dispatch.Mailbox.run(Mailbox.scala:220)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

### LINKED ISSUE #151: Provide suggested topics for subscribers

### PR DESCRIPTION
**Pull request by richardrodgers**
Fixes #151
Fixes #93

### CODE REVIEW

**Comment by JPrevost** [comment]
I'll have to refactor my PR to use a different evolution name. Shouldn't be a problem. I suspect this is why rails uses timestamps as part of a migration name instead of just an Integer.

**Comment by richardrodgers** [comment]
Ahh, sorry - I hadn't seen that I stomped on your evolution SQL

**Comment by richardrodgers** [comment]
We should also maybe open a ticket (if you agree) about coalescing all the evolutions (perhaps at each milestone) at some point.

File: test/unit/SubscriberSpec.scala [hunk 1]
```diff
@@ -2,9 +2,7 @@ import org.specs2.mutable._
 import play.api.test._
 import play.api.test.Helpers._
 
-import models.Channel
-import models.Collection
-import models.ContentType
+import models.{Agent, Channel, Collection, ContentType}
```
Comment (JPrevost) [inline test/unit/SubscriberSpec.scala]: Ah... I'll start doing this too. Listing one per line was annoying me as I was doing it :)

**Comment by JPrevost** [comment]
I'm not sure we should change the migrations after they are merged as it may lead to confusion as to what is applied where. I'm definitely fine with a ticket to consider it further though.

**Comment by JPrevost** [comment]
:+1:

**Comment by richardrodgers** [comment]
Sorry, I didn't mean to imply altering commits. Rather, I believe it is common to express the results of applying the migrations in a single DDL file (in rails, db/schema.rb, or db/structure.sql, is it?) that is in source control, and is the deployed file when installing. My question was whether we do that once at the end of the project, or have a m1.sql, m2.sql, etc

**Comment by JPrevost** [comment]
The `schema.rb` is indeed checked into source control in rails (generally), but usually the migrations are actually run as part of deploy. `schema.rb` only really comes into play when doing a `rake db:reset` which isn't a normal process when deploying as you'd lose all your data.

In essence, I think it would be nice if Play had something like `schema.rb` as it's convenient to look at for reference, but I think we should run the Evolutions as part of deploy as there isn't really any built in guarantee that the final state of the database as reflected by something like `schema.rb` actually matches the end results of all of the evolutions/migrations. The structure should end up the same, but technically you can insert data as part of evolutions/migrations that would not be reflected in something like `schema.rb` (although that's probably a bad idea).

**Comment by richardrodgers** [comment]
So you disagree with:

<snip>
6.1 What are Schema Files for?
Migrations, mighty as they may be, are not the authoritative source for your database schema. That role falls to either db/schema.rb or an SQL file which Active Record generates by examining the database. They are not designed to be edited, they just represent the current state of the database.

There is no need (and it is error prone) to deploy a new instance of an app by replaying the entire migration history. It is much simpler and faster to just load into the database a description of the current schema.

</snip>

(just randomly googled: at http://edgeguides.rubyonrails.org/active_record_migrations.html)

Just don't know rails best practices, sorry to be dense

**Comment by JPrevost** [comment]
I honestly don't know how you'd use a schema file to selectively apply changes without losing the entire database in the process.

**Pull request merged**